src/Connector/CachingConnector.php

```diff
@@ -76,6 +76,6 @@ public function isCacheEnabled()
 
     private function hash(array $structure)
     {
-        return json_encode($structure);
+        return preg_replace('/[\{\}\(\)\/\\\@:]/', '.', json_encode($structure));
```

Comment:
We should prefer `str_replace()` here because regular expressions must be compiled and are therefore typically slower to execute. Consider assigning the array of replacement characters to a local static variable to lazily allocate the replacements list once instead of for each method invocation.